pkg/pillar/cmd/watcher/watcher.go

```diff
@@ -0,0 +1,407 @@
+// Copyright (c) 2021-2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package watcher
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/utils"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	agentName      = "watcher"
+	errorTime      = 3 * time.Minute
+	warningTime    = 40 * time.Second
+	usageThreshold = 2
```

Comment:
Would be good to run some experiments to see if the +/- 1% hysteresis is sufficient. I would be inclined to pick a larger number; even 10 aka +/- 5% wouldn't seem exessive to me.

Reply:
I have a device with two app vms running this code for the last 12 hours. In the last 12 hours I have only seen these memory notifications being printed 4 times. And these prints happened only during the bootup process of EVE/Apps. +/- 2 seems to be holding good for now. May be we can adjust the threshold when we find it necessary?

Reply:
Makes sense to keep an eye (e.g., in Kibana) as we roll this out.